core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerPool.scala

```diff
@@ -119,7 +127,7 @@ class ContainerPool(childFactory: ActorRefFactory => ActorRef,
         }
         val createdContainer =
           // Is there enough space on the invoker for this action to be executed.
-          if (hasPoolSpaceFor(busyPool, r.action.limits.memory.megabytes.MB)) {
+          if (hasPoolSpaceFor(busyPool ++ prewarmedPoolForOtherKind, r.action.limits.memory.megabytes.MB)) {
```

Comment:
I'd think we explicitly do not want to have the prewarm pool survive, if other actions would benefit from using that space. After all, it's a performance optimization, not a guarantee.

Wouldn't this also be plumbed into the controller for it to not send requests down a path where they might not get executed?

I think this warrants a dev-list discussion 🤔

Reply:
Sorry, i didn't get it,
anyway i send a email to dev-list